Tasks/ServiceFabricUpdateManifests/Update-ApplicationVersions.psm1

```diff
@@ -57,20 +62,26 @@
 
                 Write-Host (Get-VstsLocString -Key SearchingApplicationType -ArgumentList $appTypeName)
 
-                $oldAppPackagePath = Join-Path $oldDropLocation $newAppPackagePath.SubString((Get-VstsTaskVariable -Name Build.SourcesDirectory -Require).Length + 1)
+                if (-not $newAppPackagePath.StartsWith($pkgArtifactPath))
```

Comment:
It is very difficult to make user understand about this new mandatory input. Can we think of any alternative without additional input. Can we parse $newAppPackagePath to check if it contains a path fragment which matches path inside $oldPackagePath. For example, if new path is C:\a\b\c\ApplicationPackage.xml then we can iterate whether old package contains any of following path:

- a\b\c\ApplicationPackage.xml
- b\c\ApplicationPackage.xml
- c\ApplicationPackage.xml
- ApplicationPackage.xml